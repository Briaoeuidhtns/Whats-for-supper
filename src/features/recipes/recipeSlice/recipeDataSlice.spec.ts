import recipeDataReducer, {
  Recipe,
  RecipeListState,
  addRecipe,
  shuffleRecipes,
} from './recipeDataSlice'

const defaultState = recipeDataReducer(undefined, { type: undefined })
const minimalInitialState: RecipeListState = { recipes: [] }

const dummyRecipe: Recipe = {
    title: 'Dummy Recipe',
    description: 'Dummy Recipe Description',
    tags: ['Dummy Tag'],
    rating: 3,
  },
  dummyRecipe2: Recipe = {
    title: 'Dummy Recipe 2',
    description: 'Dummy Recipe 2 Description',
    tags: ['Dummy Tag'],
  }

describe('recipe data reducer', () => {
  it('should handle addRecipe', () => {
    expect(
      recipeDataReducer(
        {
          ...minimalInitialState,
          recipes: [],
        },
        { type: addRecipe.type, payload: dummyRecipe }
      )
    ).toMatchObject({
      recipes: [dummyRecipe],
    })

    expect(
      recipeDataReducer(
        {
          ...minimalInitialState,
          recipes: [dummyRecipe],
        },
        { type: addRecipe.type, payload: dummyRecipe2 }
      )
    ).toMatchObject({
      recipes: [dummyRecipe, dummyRecipe2],
    })
  })

  it('should shuffle recipes', () => {
    // Using the large set of data to reduce chance of shuffle not changing order
    const shuffled = recipeDataReducer(defaultState, {
      type: shuffleRecipes.type,
      payload: 0,
    })
    expect(shuffled.recipes).not.toEqual(defaultState.recipes)

    expect(shuffled).toMatchObject({
      recipes: expect.arrayContaining(defaultState.recipes),
    })

    expect(shuffled.recipes).toHaveLength(defaultState.recipes.length)

    expect(shuffled.recipes).toMatchSnapshot()
  })
})
