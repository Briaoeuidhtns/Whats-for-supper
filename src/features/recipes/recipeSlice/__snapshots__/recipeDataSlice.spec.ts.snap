// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`recipe data reducer should shuffle recipes 1`] = `
Array [
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Chicken_Curry)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/a/a5/Chicken_Karaage_Curry%2C_Y.Izakaya%2C_Paris_001.jpg/250px-Chicken_Karaage_Curry%2C_Y.Izakaya%2C_Paris_001.jpg",
    "tags": Array [
      "Chicken",
      "Indian",
    ],
    "title": "Chicken Curry",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Deep_Dish_Pizza)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/d/d1/Pepperoni_pizza.jpg/300px-Pepperoni_pizza.jpg",
    "tags": Array [
      "Pork",
      "Beef",
      "American",
    ],
    "title": "Deep Dish Pizza",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Chicken_Biryani)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/0/0c/BurmeseBiryani.jpg/300px-BurmeseBiryani.jpg",
    "tags": Array [
      "Chicken",
      "Italian",
    ],
    "title": "Chicken Biryani",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Crispy_Chicken)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/c/c5/Burger_King_Chicken_Nuggets_%2815423233415%29.jpg/250px-Burger_King_Chicken_Nuggets_%2815423233415%29.jpg",
    "tags": Array [
      "Chicken",
      "American",
    ],
    "title": "Crispy Chicken",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Boeuf_Bourguignon)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/5/53/Boeuf_bourguignon_servi_avec_des_p%C3%A2tes.jpg/290px-Boeuf_bourguignon_servi_avec_des_p%C3%A2tes.jpg",
    "tags": Array [
      "Beef",
      "French",
    ],
    "title": "Boeuf Bourguignon",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Barbecued_Spare_Ribs)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/1/15/Spare_ribs_with_Chinese_barbecue_sauce_cropped.jpg/300px-Spare_ribs_with_Chinese_barbecue_sauce_cropped.jpg",
    "tags": Array [
      "Beef",
      "Barbeque",
    ],
    "title": "Barbecued Spare Ribs",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Frikadeller)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/0/0e/Frikadeller.jpg/300px-Frikadeller.jpg",
    "tags": Array [
      "Pork",
      "Beef",
      "German",
    ],
    "title": "Frikadeller",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Bangers_%26_Mash)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/6/68/Bangers_and_mash_1.jpg/250px-Bangers_and_mash_1.jpg",
    "tags": Array [
      "Pork",
      "British",
    ],
    "title": "Bangers & Mash",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Chow_Mein)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/d/d1/Chow_mein_1_by_yuen.jpg/300px-Chow_mein_1_by_yuen.jpg",
    "tags": Array [
      "Egg",
      "Chinese",
    ],
    "title": "Chow Mein",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Chicken_Vindaloo)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/e/e0/Chicken_Vindaloo.jpeg/300px-Chicken_Vindaloo.jpeg",
    "tags": Array [
      "Chicken",
      "Indian",
    ],
    "title": "Chicken Vindaloo",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Coronation_Chicken)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/2/20/Coronation_Chicken.jpg/250px-Coronation_Chicken.jpg",
    "tags": Array [
      "Chicken",
      "British",
    ],
    "title": "Coronation Chicken",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Arroz_con_Pollo)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/3/39/Arroz-con-Pollo.jpg/300px-Arroz-con-Pollo.jpg",
    "tags": Array [
      "Chicken",
      "Spanish",
    ],
    "title": "Arroz con Pollo",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Fleischkrapfen)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/2/24/Fleischkrapfen_on_plate.jpg/250px-Fleischkrapfen_on_plate.jpg",
    "tags": Array [
      "Pork",
      "Austrian",
    ],
    "title": "Fleischkrapfen",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Arroz_Negro)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/5/5b/Arroz_negro_02.JPG/350px-Arroz_negro_02.JPG",
    "tags": Array [
      "Squid",
      "Spanish",
    ],
    "title": "Arroz Negro",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Curry_Chicken)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/e/e4/Indian_Curry_Chicken.jpg/250px-Indian_Curry_Chicken.jpg",
    "tags": Array [
      "Chicken",
      "Indian",
    ],
    "title": "Curry Chicken",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Chicken_Tikka_Masala)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/1/13/Chicken_Tikka_Masala_Curry.png/250px-Chicken_Tikka_Masala_Curry.png",
    "tags": Array [
      "Chicken",
      "Asian",
    ],
    "title": "Chicken Tikka Masala",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Calzone)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/6/6d/20000227--calzone.jpg/300px-20000227--calzone.jpg",
    "tags": Array [
      "Pork",
      "Beef",
      "Italian",
    ],
    "title": "Calzone",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Com_Chien)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/d/d1/Com_chien.jpg/300px-Com_chien.jpg",
    "tags": Array [
      "Egg",
      "Chinese",
    ],
    "title": "Com Chien",
  },
  Object {
    "description": "[Wikibook Cookbook link](https://en.wikibooks.org/wiki/Cookbook:Bulgogi)",
    "image": "//upload.wikimedia.org/wikipedia/commons/thumb/c/c8/Korean.food-Bulgogi-01.jpg/300px-Korean.food-Bulgogi-01.jpg",
    "tags": Array [
      "Beef",
      "Korean",
    ],
    "title": "Bulgogi",
  },
]
`;
